app/repositories/GoogleAnalytics.scala

```diff
@@ -315,6 +286,94 @@ object GoogleAnalytics {
     }
   }
 
+
+  // CTA clicks functions
+
+  def getCtaClicksForCampaign(campaignId: String): Option[Map[String, Long]] = {
+
+    AnalyticsDataCache.getCampaignCtaClicksReport(campaignId) match {
+      case Hit(report) => {
+        Logger.debug(s"getting CTA clicks for campaign $campaignId - cache hit")
+        Some(report)
+      }
+      case Stale(report) => {
+        Logger.debug(s"getting CTA clicks for campaign $campaignId - cache stale spawning async refresh")
+
+        Future{
+          Logger.debug(s"async refresh of CTA clicks for campaign $campaignId")
+          fetchAndStoreCtaClicksForCampaign(campaignId)
+        } // serve stale but spawn refresh future
+        Some(report)
+      }
+      case Miss => {
+        Logger.debug(s"getting CTA clicks for campaign $campaignId - cache miss fetching sync")
+        fetchAndStoreCtaClicksForCampaign(campaignId)
```

Comment:
If you miss, this will block until the data is fetched.  Is the aim to run `getCtaClicksForCampaign` as a background job?